src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageArgs.cs

```diff
@@ -15,12 +15,14 @@ internal class ListPackageArgs
         public string Path { get; }
         public IEnumerable<PackageSource> PackageSources { get; }
         public IEnumerable<string> Frameworks { get; }
-        public bool IncludeOutdated { get; }
-        public bool IncludeDeprecated { get; }
+        public bool OutdatedReport { get; }
+        public bool DeprecatedReport { get; }
+        public bool VulnerableReport { get; }
+        public bool IsOffline { get; }
         public bool IncludeTransitive { get; }
-        public bool Prerelease { get; }
-        public bool HighestPatch { get; }
-        public bool HighestMinor { get; }
+        public bool Prerelease { get; internal set; }
```

Comment:
why the internal set? 
We should treat the args as records.

Reply:
The `internal set` is for adjustments after creation. I had the choice of putting the "business logic" inside the class here and resetting these options if this is a `--vulnerable` report, but chose to include all of the logic in the list command so we can see it in one place.
As for the nature of the logic (whether to call these option conflicts or simp0ly ignore them), I've gone with the spec on that one.

Reply:
> The internal set is for adjustments after creation

I think the argument would be that we don't want that. 

We're adopting a functional approach into C#, but what we're hoping to achieve here is more predictability in the Runner/Command pattern we have throughout our codebase. 

All that to say, immutability was deliberate and I think we should keep it that way.

Reply:
Can do. Thanks.